### ISSUE TITLE
Rspress Roadmap

### ISSUE THREAD

**Issue by sanyuan0704** [web-infra-dev/rspress#133]
### What problem does this feature solve?

xx

### What does the proposed API look like?

xx

**Comment by jeffwcx** [comment]
Are there any plans to add component display features, similar to storybooks?

**Comment by sanyuan0704** [comment]
> Are there any plans to add component display features, similar to storybooks?

Yeah, Rspress has official preview and playground plugin for component development scene.

https://rspress.dev/plugin/official-plugins/preview
https://rspress.dev/plugin/official-plugins/playground

And if you have any need else,  welcome to feedback,

**Comment by jeffwcx** [comment]
I have installed and experienced the writing of the demo. It is very fast and the experience is very good, but there are still some points that can be enhanced.
1. Lack of CodeSandbox, Stackblitz, CodePen support.
2. Only supports react component. It is recommended to support cross-framework component display capabilities, such as Vue components.
3. The ability to extract information from react/vue components and generate API tables

**Comment by sanyuan0704** [comment]
> I have installed and experienced the writing of the demo. It is very fast and the experience is very good, but there are still some points that can be enhanced.
> 
> 1. Lack of CodeSandbox, Stackblitz, CodePen support.
> 2. Only supports react component. It is recommended to support cross-framework component display capabilities, such as Vue components.
> 3. The ability to extract information from react/vue components and generate API tables

Good advice! I will response to the three points:
1. The implement of component display in Rspress does not depend on any external sandbox such as `CodeSandbox`、`Stackblitz`, etc. But we will also support the link to jump to the corresponding platform in the future. cc @10Derozan 
2. The custom component render process is based on React as well as components in the theme.So we have no plan to support vue component in preview/playground plugin.
3. The ability is under construction and we have integrated `react-docgen-typescript` and `documentation` to generate API Table automatically.Stay tuned!

**Comment by jeffwcx** [comment]
For the second point, maybe you can run instances of other frameworks in the react framework. Storybook also uses this pattern. Maybe I can make a PR later;
For the third point, it would be better to have an interface to connect to other frameworks.

**Comment by zoolsher** [comment]
@jeffwcx Having cross-framework support as a feature in the framework would be fantastic! looking forward to your PR

**Comment by Yukiniro** [comment]
Hey guys, what should I do if I want to contribute something such as the badge component for the roadmap?

**Comment by sanyuan0704** [comment]
#542

**Comment by sanyuan0704** [comment]
> Hey guys, what should I do if I want to contribute something such as the badge component for the roadmap?

You can first put a issue and link the roadmap. Then you can implement it by pull request and we will review your code as soon as possible.

**Comment by iicdii** [comment]
> Provide google-analytics plugin to integrate Google Analytics into your site to collect site visit data.

Isn't it already available with [rsbuild-plugin-google-analytics](https://github.com/rspack-contrib/rsbuild-plugin-google-analytics)?

**Comment by chenjiahan** [comment]
> Isn't it already available with [rsbuild-plugin-google-analytics](https://github.com/rspack-contrib/rsbuild-plugin-google-analytics)?

Thanks, we has updated the roadmap status.

**Comment by jxh150535011** [comment]
> For the second point, maybe you can run instances of other frameworks in the react framework. Storybook also uses this pattern. Maybe I can make a PR later;
> For the third point, it would be better to have an interface to connect to other frameworks.

Following your approach, we have implemented the `rspress-plugin-vue` component to support the Vue framework

**Comment by chenjiahan** [comment]
Awesome 👍 you can add it to: https://github.com/web-infra-dev/awesome-rspack?tab=readme-ov-file

**Comment by SoonIter** [comment]
Closing this issue as it has been inactive for over a year. If this is still relevant, please feel free to reopen it or create a new issue with updated context. Thank you for your contribution!
